### PR TITLE
Remove C++ components

### PR DESCRIPTION
**Pull request by jl-wynen**
Please check that you can make a local installation!

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -71,36 +105,44 @@ jobs:
           name: html
           path: html/
 
-  upload_packages:
-    name: Deploy packages
-    needs: [docs]
+  upload_pypi:
```
Comment (SimonHeybrock) [inline .github/workflows/release.yml:108]: You are keeping the docs step before the package uploads?

File: docs/about/release-notes.rst [hunk 2]
```diff
@@ -34,6 +34,31 @@ Release Notes
...
+
+Features
+~~~~~~~~
+
+* ScippNeutron is now available on PyPI `#384 <https://github.com/scipp/scipp/pull/384>`_.
+
+Breaking changes
+~~~~~~~~~~~~~~~~
+
+* Removed C++ components and CMake package configuration. This only affects users that linked against the C++ part of ScippNeutron`#384 <https://github.com/scipp/scipp/pull/384>`_.
```
Comment (SimonHeybrock) [inline docs/about/release-notes.rst:48]:
```suggestion
* Removed C++ components and CMake package configuration. This only affects users that linked against the C++ part of ScippNeutron `#384 <https://github.com/scipp/scipp/pull/384>`_.
```

File: pyproject.toml [hunk 3]
```diff
@@ -1,5 +1,54 @@
...
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+requires-python = ">=3.8"
+dynamic = ["version"]
+dependencies = [
+    "h5py",
+    "numpy>=1.20",
+    "pooch",
```
Comment (SimonHeybrock) [inline pyproject.toml:32]: Wondering if we really need to require `pooch`?

File: tox.ini [hunk 4]
```diff
@@ -1,17 +1,44 @@
...
-skip_install = true
+deps = -r requirements/test.txt
+setenv =
+  JUPYTER_PLATFORM_DIRS = 1
+commands = pytest
+
+[testenv:docs]
+description = invoke sphinx-build to build the HTML docs
+basepython = python3.8
+deps = {posargs:}
```
Comment (SimonHeybrock) [inline tox.ini:14]: No harm in keeping this, but the posargs where used to set the version in the docs workflow (to install via pip), which I think you are not using for ScippNeutron?

**Review by SimonHeybrock — COMMENTED** [review]
Nice, must have been satisfying to delete so many things. :)

File: .github/workflows/release.yml (hunk 1 above)
Comment (jl-wynen) [reply .github/workflows/release.yml:108]: This PR already changes a lot about the build process. I want to keep it as simple as possible and update other aspects separately.

File: pyproject.toml (hunk 3 above)
Comment (jl-wynen) [reply pyproject.toml:32]: We can split off a few into an extra.

File: tox.ini (hunk 4 above)
Comment (jl-wynen) [reply tox.ini:14]: But won't we do that in the future?

Comment (SimonHeybrock) [reply tox.ini:14]: See above, not sure what your plan is regarding the Mantid dependency.

File: .github/workflows/release.yml (hunk 1 above)
Comment (jl-wynen) [reply .github/workflows/release.yml:108]: I am planning to change it. But concerning Mantid, we can't use tox unless we allow it to access external packages. At this point we might get into trouble with conflicting conda and pip packages.

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**